elastic/kibana
# Add context.originalAlertState to the Metric Threshold and Inventory Threshold recovery context

## Summary

This PR adds the `ALERT_ACTION_GROUP` to the Alerts-As-Data documents for both the Metric Threshold and Inventory Threshold rules. It then uses that value from the alert document in the recovery context to set `context.originalAlertState`. This also adds `context.originalStateWasALERT`, `context.originalStateWasWARNING`, and `context.originalStateWasNO_DATA` (Metric Threshold Only) to allow for conditional Mustache templates. I also fixed the types for `getAlertByAlertUuid()` to be more accurate.

#### Metric Threshold Example
```
{{#context.originalAlertStateWasALERT}}
This is a recovery for an ALERT
{{/context.originalAlertStateWasALERT}}

{{#context.originalAlertStateWasWARNING}}
This is a recovery for a WARNING
{{/context.originalAlertStateWasWARNING}}

{{#context.originalAlertStateWasNO_DATA}}
This is a recovery for NO_DATA
{{/context.originalAlertStateWasNO_DATA}}
```

#### Inventory Threshold Example
```
{{#context.originalAlertStateWasALERT}}
This is a recovery for an ALERT
{{/context.originalAlertStateWasALERT}}

{{#context.originalAlertStateWasWARNING}}
This is a recovery for a WARNING
{{/context.originalAlertStateWasWARNING}}
```
### How to test

1. Start Kibana and ingest some data (Metricbeat or whatever)
2. Create a rule (one for each), for the Metric Threshold rule you will need to group by something like `host.name`
3. Set the conditions to something you can trigger, I used `NO_DATA`
4. Add a server log action for the recovery action group with `{{context}}`, alternatively you can use the examples above to see the Mustache logic work
5. Save the rules
6. Stop ingesting data and allow the rule to trigger a `NO DATA` alert
7. Start ingesting data so that it recovers
8. Observe the log message with `originalAlertState` as `NO DATA` for Metric Threshold and `ALERT` for Inventory Threshold.
